dhis2/cli-utils-cypress
# feat: remove cli util, update cypress to v12 & small conceptual adjustments

* No more `cy.login` (now there's a `cy.fillInLoginForm` command which can be used together with `cy.session`)
* The `d2-cli-cypress` tool is gone, it's not needed anymore (it is actually still there (the `install` command), but does nothing; as discussed with @HendrikThePendric)
* The commands & plugins libraries now have updated/correct peer dependencies
* Example apps now use cypress@12 & the latest versions of the commands / plugins
* Docs have been updated

## How to review this PR

There are several things that should be checked:

* Read the docs and ensure they're correct & understandable. I've changed quite a bit and I'd like the docs to be as good as possible. If there's something unclear, e.g. some context is missing, please let me know. In this case, too much is better than too little information.
* Ideally you use an app that has cypress test and follow the upgrade guide of the docs. That should ensure that the docs are complete & correct as well as that the libraries work as intended
* Repeat the steps I've listed below, under "How to test"
* Check if the `d2-utils-cypress install` command does nothing, just informing the user that the command is not supported anymore

## How to test

### Example apps

* Check out this branch and install dependencies
* Run `yarn test:examples` (will start the example apps & run cypress tests, which includes both capturing and stubbing in the example app that uses the network shim)

### Real app

* Check out this branch and install dependencies
* Build the commands library (`yarn workspace @dhis2/cypress-commands build`)
* Link the commands & plugins libraries (`yarn workspace @dhis2/cypress-commands link && yarn workspace @dhis2/cypress-plugins link`)
* Check out the `upgrade-cypress-to-v12` branch of the aggregate data entry app (related PR: dhis2/aggregate-data-entry-app#319)
* Add the linked dependencies (`yarn link @dhis2/cypress-plugins @dhis2/cypress-commands`)
* Start the aggregate data entry app (`yarn cypress:start`)
* Play around with the UI (`yarn cypress:open:live`) and/or capturing/stubbing network requests (`yarn cypress:run:capture && yarn cypress:run:stub`)
